x-pack/solutions/search/plugins/search_indices/public/hooks/api/use_indices_status.ts

```diff
@@ -15,14 +18,20 @@ import { useKibana } from '../use_kibana';
 
 const DEFAULT_INDICES_POLLING_INTERVAL = 15 * 1000;
 
-export const useIndicesStatusQuery = (pollingInterval = DEFAULT_INDICES_POLLING_INTERVAL) => {
+export const useIndicesStatusQuery = (
+  queryClient?: QueryClient,
+  pollingInterval = DEFAULT_INDICES_POLLING_INTERVAL
+): UseQueryResult<IndicesStatusResponse> => {
   const { http } = useKibana().services;
+  const clientContext = createContext(queryClient);
+
   return useQuery({
     refetchInterval: pollingInterval,
     refetchIntervalInBackground: true,
     refetchOnWindowFocus: 'always',
     retry: true,
     queryKey: [QueryKeys.FetchSearchIndicesStatus],
     queryFn: () => http.get<IndicesStatusResponse>(GET_STATUS_ROUTE),
+    context: queryClient ? clientContext : undefined,
```

Comment:
```suggestion
  const clientContext = queryClient ? createContext(queryClient) : undefined;

  return useQuery({
    refetchInterval: pollingInterval,
    refetchIntervalInBackground: true,
    refetchOnWindowFocus: 'always',
    retry: true,
    queryKey: [QueryKeys.FetchSearchIndicesStatus],
    queryFn: () => http.get<IndicesStatusResponse>(GET_STATUS_ROUTE),
    context: clientContext,
```

I feel like we should probably also `useMemo` the `clientContext` but I'm more curious why we need it and how doing this effects the usage of this outside of the `enterprise_search` overview page. I assume this is to handle sharing this hook and using it where the query client provider is not available?

Reply:
`I assume this is to handle sharing this hook and using it where the query client provider is not available?`
Yeah, we need it if we want to share hook where queryProvider is not init. 
But if we a going to use just a fetch request, then we can remove this changes